packages/toolpad-app/src/toolpad/AppState.tsx

```diff
@@ -17,18 +17,21 @@ import { hasFieldFocus } from '../utils/fields';
 import { DomView, getViewFromPathname } from '../utils/domView';
 import config from '../config';
 
+export function getNodeHashes(dom: appDom.AppDom): NodeHashes {
+  return mapValues(dom.nodes, (node) => insecureHash(JSON.stringify(node)));
+}
+
 export type ComponentPanelTab = 'component' | 'theme';
 
-export type DomAction =
```

Comment:
It might be better to just see how this file would look now instead of just looking at the diff.